chiselFrontend/src/main/scala/chisel3/core/Module.scala

```diff
@@ -49,7 +49,7 @@ object Module {
     val module: T = bc  // bc is actually evaluated here
 
     if (Builder.whenDepth != 0) {
-      throwException("Internal Error! When depth is != 0, this should not be possible")
+      throwException("Internal Error! \"When\" scope depth is != 0, this should have been caught!")
```

Comment:
If `"When"` is referring to the construct (unless I misunderstood it), then I would recommend `when() scope depth is != 0; this should have been caught!`.

Reply:
Done